pkg/metrics/metrics.go

```diff
@@ -126,27 +184,71 @@ func (opMgr *operationMetricsManager) RecordMetrics(op Operation, status Operati
 		opMgr.cache.Delete(op)
 		return
 	}
-	strStatus := opStatusUnknown
+	strStatus := string(SnapshotStatusTypeUnknown)
 	if status != nil {
 		strStatus = status.String()
 	}
-	duration := time.Since(ts).Seconds()
-	opMgr.opLatencyMetrics.WithLabelValues(op.Driver, op.Name, strStatus).Observe(duration)
+
+	// record invidual reconciliation time
+	reconciliationDuration := time.Since(ts).Seconds()
+	opMgr.opReconciliationLatencyMetrics.WithLabelValues(op.Driver, op.Name, op.SnapshotType, strStatus).Observe(reconciliationDuration)
+
+	// record total operation durations if operation is successful
+	if strStatus == string(SnapshotStatusTypeSuccess) {
+		switch op.Name {
+		case CreateSnapshotAndReadyOperationName:
```

Comment:
does tracking e2e time for CreateSnapshot also seem useful?

Reply:
good point, added this